File: Unstop/100DayCode/Day 5/BasketBall.py
def calculate_final_score(ops):
    scores = []
    
    for op in ops:
        if op == "C":
            if scores:
                scores.pop()
        elif op == "D":
            if scores:
                scores.append(2 * scores[-1])
        elif op == "+":
            if len(scores) >= 2:
                scores.append(scores[-1] + scores[-2])
        else:
            scores.append(int(op))
    
    return sum(scores)

# Reading input
n = int(input().strip())
ops = input().strip().split()

# Calculating and printing the final score
result = calculate_final_score(ops)
print(result)

File: Unstop/100DayCode/Day 5/BasketBall.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int calculateFinalScore(vector<string>& ops) {
    vector<int> scores;

    for (const string& op : ops) {
        if (op == "C") {
            if (!scores.empty()) {
                scores.pop_back();
            }
        } else if (op == "D") {
            if (!scores.empty()) {
                scores.push_back(2 * scores.back());
            }
        } else if (op == "+") {
            if (scores.size() >= 2) {
                scores.push_back(scores[scores.size() - 1] + scores[scores.size() - 2]);
            }
        } else {
            scores.push_back(stoi(op));
        }
    }

    int sum = 0;
    for (int score : scores) {
        sum += score;
    }

    return sum;
}

int main() {
    int n;
    cin >> n;
    vector<string> ops(n);
    
    for (int i = 0; i < n; ++i) {
        cin >> ops[i];
    }

    int result = calculateFinalScore(ops);
    cout << result << endl;

    return 0;
}
